src/viam/sdk/common/exception.cpp

```diff
@@ -0,0 +1,37 @@
+#include <viam/sdk/common/exception.hpp>
+
+namespace viam {
+namespace sdk {
+
+ViamException::ViamException(const std::string& what)
+    : std::runtime_error("ViamException: " + what){};
+
+ViamException::ViamException(const std::string& what, const std::string& type)
+    : std::runtime_error("ViamException(" + type + "): " + what){};
+
+ViamException::~ViamException() = default;
+
+ConnectionException::ConnectionException(const std::string& what)
+    : ViamException(what, "Connection"){};
+
+DuplicateRegistrationException::DuplicateRegistrationException(const std::string& what)
+    : ViamException(what, "DuplicateRegistration"){};
+
+DuplicateResourceException::DuplicateResourceException(const std::string& what)
+    : ViamException(what, "DuplicateResource"){};
+
+GRPCException::GRPCException(const std::string& what, int grpc_error_code)
+    : ViamException(what, "gRPC"), grpc_error_code_(std::move(grpc_error_code)){};
```

Comment:
Since `grpc_error_code` is an int (which doesn't have a move constructor), I think this `std::move` doesn't do anything.